onnxruntime/core/providers/ngraph/ngraph_execution_provider.cc

```diff
@@ -512,14 +567,19 @@ static ONNX_NAMESPACE::ModelProto GetModelProtoFromFusedNode(const onnxruntime::
 Status NGRAPHExecutionProvider::Compile(const std::vector<onnxruntime::Node*>& fused_nodes,
                                         std::vector<NodeComputeInfo>& node_compute_funcs) {
   for (const auto& fused_node : fused_nodes) {
-    auto model_proto = GetModelProtoFromFusedNode(fused_node);
-
     NodeComputeInfo compute_info;
 
+    const auto& attributes = fused_node->GetAttributes();
+    const auto graph_name = attributes.at("graph_name").s();
+    // find the shared_ptr with initializes of a graph this fused_node belongs to
+    auto initializers = graph_initializers_.at(graph_name);
+
     // Local copy of backend since, class members cannot be captured.
     auto ngraph_backend = ng_backend_;
-    compute_info.create_state_func = [model_proto, ngraph_backend](ComputeContext* context, FunctionState* state) {
-      auto* p = new onnxruntime::ngraph_ep::NGRAPHCustomOp(context, model_proto, ngraph_backend);
+    compute_info.create_state_func = [model_proto = GetModelProtoFromFusedNode(fused_node), ngraph_backend, initializers]
```

Comment:
one thing I wanted to understand a little better is whether we can construct the model_proto from the fused_node without the specialized function (GetModelProtoFromFusedNode). 
if we look at TensorRT execution provider, it seems possible.
see:  https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L310
to
https://github.com/microsoft/onnxruntime/blob/master/onnxruntime/core/providers/tensorrt/tensorrt_execution_provider.cc#L317

TRT requires the model_proto with initializers when constructing the engine so I assume it is working.

When we construct the meta_def in GetCapability() for the Subgraph, the inputs should contain the initializers
so when ORT framework does the node fusion, the fused node should have the initializers. 
I was hoping we wouldn't need to keep the initializers separately. 

+@linkerzhang to confirm

Reply:
Thank you for the details. I'm working on it now to see if I can re-use this approach in our provider.

Reply:
I've created a separate branch based on the one from this PR. The top-most commit contains the changes that I made towards creating `ModelProto` from a `fused_node` in the `Compile` method. Please have a look and let me know if I should merge it into this PR.

branch: https://github.com/NervanaSystems/onnxruntime/tree/tomdol/memfix_fused_node
diff: https://github.com/NervanaSystems/onnxruntime/commit/7addba7b9bbfdc6792d7aee42c3f6272b6523cd9

Reply:
> I've created a separate branch based on the one from this PR. The top-most commit contains the changes that I made towards creating `ModelProto` from a `fused_node` in the `Compile` method. Could you please verify if the memory consumption numbers look better with this change?
> 
> branch: https://github.com/NervanaSystems/onnxruntime/tree/tomdol/memfix_fused_node
> diff: [NervanaSystems@7addba7](https://github.com/NervanaSystems/onnxruntime/commit/7addba7b9bbfdc6792d7aee42c3f6272b6523cd9)


we'll take a look. thanks.

Reply:
Thanks.
we see same memory footprint improvement for both tomdol/ngraph_update and tomdol/memfix_fused_node 
I think that should be expected, right?
In terms of code organization, tomdol/memfix_fused_node should be the better approach.

btw, I recall you guys have perf regression tests. I assume everything looks good with these new changes?

Reply:
I see the same results too. I was thinking if this could potentially save a bit more for resnet but it didn't. But the code is simpler and I think this is the better way to construct the subgraph's ModelProto. I'm going to merge it into this PR's branch.

I've received the performance measurements yesterday (done with this PR branch) and everything looks stable.

Thanks again for the feedback :)